### PR TITLE
Add session "Embrace Swift type inference"

### PR DESCRIPTION
**Pull request by rogerluan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -551,4 +551,29 @@ _Jonathan Hersh, Allen Langmaier_
 - **Updates to development**
   - Dynamic updates to list elements
   - `CPListItem`
-  - `CPListImageRowItem`
\ No newline at end of file
+  - `CPListImageRowItem`
+
+## Embrace Swift type inference
```
Comment (Blackjacx) [inline README.md]: You forgot the TOX entry for this one, could not happen anymore after I merge your prefill-sessions PR, which is awesome btw. 🎉

File: README.md [hunk 2]
```diff
@@ -551,4 +551,29 @@ _Jonathan Hersh, Allen Langmaier_
...
+	- **Type inference errors**
+		- If a puzzle can't be solved, there's an error in the source code
+		- During type inference, the compiler will record information about errors in source code
+		- Compiler uses heuristics to attempt to fix errors in order to continue type inference
+		- Once type inference is done, the compiler will provide actionable error messages based on collected information (some errors might have auto-fixes)
+	- Swift 5.3 and Xcode 12 improved error handling a lot, showing more meaningful and reliable error messages
+- **Using Swift and Xcode to fix compile errors**
+	- The compiler will leave breadcrumbs about what the compiler was doing when it found an error
+	- This can help you connect the dots between the error you're seeing on the editor and other files in your project
+	- Hold `⌥ (Option) + ⇧ (Shift)` on an error breadcrumb and drag it to the right of the source editor, to open the source editor and the breadcrumbs details side by side
```
Comment (Blackjacx) [inline README.md]: I wound that code-style text makes symbols very small, this is fixed by bold text :-) 

```suggestion
	- Hold **⌥** (Option) **+ ⇧** (Shift) on an error breadcrumb and drag it to the right of the source editor, to open the source editor and the breadcrumbs details side by side
```

**Review by Blackjacx — CHANGES_REQUESTED** [review]
Thanks for the addition. I love it when it becomes even hard to follow up with the PRs ^^

**Comment by Blackjacx** [comment]
Would be cool if you could rebase your PR and put the session in the correct position :)

**Comment by rogerluan** [comment]
Done @Blackjacx 👍

**Review by Blackjacx — APPROVED** [review]
Thanks for also contributing the more technical ones 🙏
